common/app/common/configuration.scala

```diff
@@ -1,33 +1,76 @@
+import com.gu.cm.{PlayDefaultLogger, ClassPathConfigurationSource, FileConfigurationSource, Logger}
+import com.typesafe.config.ConfigException
 import conf.switches.Switches
 import conf.{Configuration, Static}
 import org.apache.commons.io.IOUtils
 import play.api.Play
 
-import scala.util.Try
+import scala.collection.JavaConversions._
+import scala.util.{Failure, Success, Try}
 
 class BadConfigurationException(msg: String) extends RuntimeException(msg)
 
-class GuardianConfiguration extends Logging {
-  implicit private lazy val app = Play.current
+object GuardianConfiguration extends Logging {
 
-  case class OAuthCredentials(oauthClientId: String, oauthSecret: String, oauthCallback: String)
-  case class OAuthCredentialsWithMultipleCallbacks(oauthClientId: String, oauthSecret: String, authorizedOauthCallbacks: List[String])
+  import com.gu.cm.{Configuration => CM}
+  import com.typesafe.config.Config
+  lazy val configuration = {
 
-  protected val configuration = ConfigurationFactory.getNonLoggingConfiguration("frontend", "env")
+    val stage = {
+      val p = new JavaProperties()
+      p.load(new FileInputStream(s"/etc/gu/install_vars"))
+      p.getProperty("STAGE", "DEV")
+    }
+    lazy val userPrivate = FileConfigurationSource(s"${System.getProperty("user.home")}/.gu/frontend.properties")
+    lazy val opsPrivate = FileConfigurationSource(s"/etc/gu/frontend.properties")
+    lazy val public = ClassPathConfigurationSource(s"env/$stage.properties")
+    new CM(List(userPrivate, opsPrivate, public), PlayDefaultLogger).load
+  }
+
+  implicit class ScalaConvertProperties(conf: Config) {
+
+    def getStringProperty = getProperty(conf.getString)_
+    def getMandatoryStringProperty = getMandatoryProperty(conf.getString)_
+    def getIntegerProperty = getProperty(conf.getInt)_
+
+    def getPropertyNames: Seq[String] = conf.entrySet.toSet.map((_.getKey): Entry[String, _] => String).toSeq
+    def getStringPropertiesSplitByComma(propertyName: String): List[String] = {
+      getStringProperty(propertyName) match {
+        case Some(property) => (property split ",").toList
+        case None => Nil
+      }
+    }
 
-  private implicit class OptionalString2MandatoryString(conf: com.gu.conf.Configuration) {
-    def getMandatoryStringProperty(property: String) = configuration.getStringProperty(property)
+    def getMandatoryProperty[T](get: String => T)(property: String) = getProperty(get)(property)
       .getOrElse(throw new BadConfigurationException(s"$property not configured"))
+    def getProperty[T](get: String => T)(property: String): Option[T] = Try(get(property)) match {
+      case Success(value) => Some(value)
+      case Failure(e: ConfigException.Missing) => None
+      case Failure(e) =>
+        log.error(s"couldn't retrive $property", e)
+        None
+    }
+
   }
 
+}
+
+class GuardianConfiguration extends Logging {
+  import GuardianConfiguration._
+  implicit private lazy val app = Play.current
```

Comment:
I am in the process to remove all reference to the global object `Play.current` which is deprecated in Play 2.5.
Can you inject it as a class parameter? Thx